src/rgw/rgw_metadata.cc

```diff
@@ -174,6 +174,17 @@ class RGWMetadataLogOmap final : public RGWMetadataLogBE {
     const cls_log_header empty_info;
     *exists = false;
     *has_entries = false;
+    static bool check_if_exists = false;
```

Comment:
Take a look at #38230 , this removes the opspam and factors out the check/upgrade logic.

Reply:
Thanks Adam. Will remove these changes (last commit). I have one doubt though. RGWMetadataLog objects are created and initialized not just during startup, but also while handling  RGWOp_MDLog_* fops runtime. So my intention was to avoid checking if FIFO/Omap exists in those cases as well. 

I think #38230 PR handles that too. Could you please confirm.

Reply:
Yes. The quick check and LazyFIFO should handle that.